### ISSUE TITLE
Wrong line number for invalid iterable that uses __getattr__

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#14892]
This generates an error ~~false positive~~ which is reported on the wrong line ~~so there are likely two bugs~~:
```py
from typing import Union, Tuple

class C:
    def __getattr__(self, name):
        pass

class D:
    def f(self) -> C: # "C" has no attribute "__iter__" (not iterable)
        return C()

    def g(self) -> None:
        a, b = self.f()  # This is where the error originates from
```
~~`C` should be treated as iterable, since it defines `__getattr__`.~~

**Comment by AlexWaygood** [comment]
> `C` should be treated as iterable, since it defines `__getattr__`.

Is that correct? At runtime, dunders are looked up on the class object rather than the instance:

```pycon
>>> class Foo:
...     def __getattr__(self, name):
...         if name == "__iter__":
...             return iter(range(10))
...
>>> Foo().__iter__
<range_iterator object at 0x00000194BCD3BB50>
>>> list(Foo().__iter__)
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
>>> list(Foo())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'Foo' object is not iterable
```

**Comment by AlexWaygood** [comment]
Using the original example:

```pycon
>>> class C:
...     def __getattr__(self, name):
...         if name == "__iter__":
...             return iter(range(2))
...
>>> class D:
...     def f(self):
...         return C()
...     def g(self):
...         a, b = self.f()
...
>>> D().g()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 5, in g
TypeError: cannot unpack non-iterable C object
```
